torch/onnx/_internal/exporter/_core.py

```diff
@@ -126,13 +126,17 @@ def tobytes(self) -> bytes:
         # it avoids copying to a NumPy array
         import torch._subclasses.fake_tensor
 
-        if isinstance(self.raw, torch._subclasses.fake_tensor.FakeTensor):
+        with torch._subclasses.fake_tensor.unset_fake_temporarily():
+            # Disable any fake mode so calling detach() etc. will return a real tensor
```

Comment:
After doing this, does it mean the line 133 will be False forever?

Reply:
Unless the original is fake already

Reply:
Could you please add a test in test_api.py for this?

Reply:
Added tests